Code/server/GameServer.cpp

```diff
@@ -383,6 +391,129 @@ void GameServer::BindServerCommands()
                 out->error("Day must be between 0 and 31, month must be between 0 and 11, and year must be between 0 and 999.");
             }
         });
+
+    m_commands.RegisterCommand<std::string>(
+        "AddAdmin", "Add admin privileges from player", [&](Console::ArgStack& aStack) {
+            auto out = spdlog::get("ConOut");
+
+            const auto& cUsername = aStack.Pop<String>();
+
+            bool playerFound = false;
+
+            PlayerManager::Get()->ForEach([&](const Player* apPlayer) {
+                if (apPlayer->GetUsername() == cUsername)
+                {
+                    AddAdminSession(apPlayer->GetConnectionId());
+                    out->info("{} admin privileges added", cUsername.c_str());
+                    playerFound = true;
+                    return;
+                }
+            });
+
+            if (!playerFound)
+            {
+                // space in username handling
+                String backupUsername = cUsername;
+                if (cUsername.find('_') != std::string::npos)
+                {
+
+                    while (backupUsername.find('_') != std::string::npos)
+                    {
+                        std::ranges::replace(backupUsername, '_', ' ');
+                    }
+
+                    PlayerManager::Get()->ForEach([&](const Player* apPlayer) {
+                        if (apPlayer->GetUsername() == backupUsername)
+                        {
+                            AddAdminSession(apPlayer->GetConnectionId());
+                            out->info("{} admin privileges added", backupUsername.c_str());
+                            playerFound = true;
+                            return;
+                        }
+                    });
+
+                }
+                if (!playerFound)
+                    out->warn("{} is not an admin", backupUsername.c_str());
+            }
+        });
+    m_commands.RegisterCommand<std::string>(
+        "RemoveAdmin", "Remove admin privileges from player", [&](Console::ArgStack& aStack) {
+            auto out = spdlog::get("ConOut");
+
+            const auto& cUsername = aStack.Pop<String>();
+            bool playerFound = false;
+
+            for (const auto& cAdmin : m_adminSessions)
+            {
+                Player* pPlayer = PlayerManager::Get()->GetByConnectionId(cAdmin);
```

Comment:
Probably the get user by name stuff too. maybe have a fn there to get a user by name.

Reply:
Added two things for this, GameServer::GetAdminByUsername and PlayerManager::GetByUsername
Reworked that part to adapt it in as well